api/src/test/java/org/openmrs/api/handler/ExistingOrNewVisitAssignmentHandlerTest.java

```diff
@@ -155,4 +155,75 @@ public void beforeCreateEncounter_shouldResolveEncounterAndVisitTypeUuidsAsGloba
 		                .getVisitType());
 	}
+
+	/**
+	 * @see org.openmrs.api.handler.ExistingVisitAssignmentHandler#beforeCreateEncounter(Encounter)
+	 */
+	@Test
+	public void beforeCreateEncounter_shouldResolveDefaultWildCardEncounterTypeAndVisitTypeIdMappingGlobalPropertyValues() {
+		Encounter encounter = Context.getEncounterService().getEncounter(1);
+		assertNull(encounter.getVisit());
+
+		Calendar calendar = Calendar.getInstance();
+		calendar.setTime(encounter.getEncounterDatetime());
+		calendar.set(Calendar.YEAR, 1900);
+
+		encounter.setEncounterDatetime(calendar.getTime());
+
+		GlobalProperty gp = new GlobalProperty(OpenmrsConstants.GP_ENCOUNTER_TYPE_TO_VISIT_TYPE_MAPPING, "default:2");
+		Context.getAdministrationService().saveGlobalProperty(gp);
+
+		new ExistingOrNewVisitAssignmentHandler().beforeCreateEncounter(encounter);
+
+		assertNotNull(encounter.getVisit());
+		assertEquals(Context.getVisitService().getVisitType(2), encounter.getVisit().getVisitType());
+	}
+	
+	/**
+	 * @see org.openmrs.api.handler.ExistingVisitAssignmentHandler#beforeCreateEncounter(Encounter)
+	 */
+	@Test
+	public void beforeCreateEncounter_shouldResolveDefaultWildCardEncounterTypeAndVisitTypeUuidMappingGlobalPropertyValues() {
+		Encounter encounter = Context.getEncounterService().getEncounter(1);
+		assertNull(encounter.getVisit());
+
+		Calendar calendar = Calendar.getInstance();
+		calendar.setTime(encounter.getEncounterDatetime());
+		calendar.set(Calendar.YEAR, 1900);
+
+		encounter.setEncounterDatetime(calendar.getTime());
+		
+		GlobalProperty gp = new GlobalProperty(OpenmrsConstants.GP_ENCOUNTER_TYPE_TO_VISIT_TYPE_MAPPING,
+			"default:c0c579b0-8e59-401d-8a4a-976a0b183519");
+		Context.getAdministrationService().saveGlobalProperty(gp);
+
+		new ExistingOrNewVisitAssignmentHandler().beforeCreateEncounter(encounter);
+
+		assertNotNull(encounter.getVisit());
+		assertEquals(Context.getVisitService().getVisitType(1), encounter.getVisit().getVisitType());
+	}
+	
+	/**
+	 * @see org.openmrs.api.handler.ExistingVisitAssignmentHandler#beforeCreateEncounter(Encounter)
+	 */
+	@Test
+	public void beforeCreateEncounter_shouldResolveDefaultWildCardEncounterTypeMappingsOverrideGlobalPropertyValues() {
+		Encounter encounter = Context.getEncounterService().getEncounter(1);
+		assertNull(encounter.getVisit());
+
+		Calendar calendar = Calendar.getInstance();
+		calendar.setTime(encounter.getEncounterDatetime());
+		calendar.set(Calendar.YEAR, 1900);
+
+		encounter.setEncounterDatetime(calendar.getTime());
+		
+		GlobalProperty gp = new GlobalProperty(OpenmrsConstants.GP_ENCOUNTER_TYPE_TO_VISIT_TYPE_MAPPING,
+			"3:4, 5:2, default:c0c579b0-8e59-401d-8a4a-976a0b183519, 2:2");
+		Context.getAdministrationService().saveGlobalProperty(gp);
+
+		new ExistingOrNewVisitAssignmentHandler().beforeCreateEncounter(encounter);
+
+		assertNotNull(encounter.getVisit());
+		assertEquals(Context.getVisitService().getVisitType(1), encounter.getVisit().getVisitType());
```

Comment:
I do not see any visitTypeId value of `1` in the above global property. Do you mind explaining what you are trying to test here?

Reply:
Do you mind explaining this a bit more?

Reply:
Provided a gp with overridden mappings which also included the `default` wildcard `EncounterType` with the valid `VisitType` uuid to test if `beforeCreateEncounter()` can resolve the `default` wildcard if it is provided in an overridden gp mapping.